pkg/agent/bakerapi.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package agent
+
+import (
+	"context"
+
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+)
+
+type AgentBaker interface {
```

Comment:
does the agentbaker pkg itself need to define the interface contract that it implements? In go it's more common to define the interface at the consuming pkg (i.e our backend rp code)

Reply:
In our code base, do we have an example with interface being defined at the consuming side? I saw more examples with interfaces defined at the library code. For example, our core/clients defined all the interfaces. Another example is the Azure SDK for Go library we use: it defines the interfaces: https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2021-03-01/compute/computeapi/interfaces.go